chainermn/communicators/_memory_utility.py

```diff
@@ -112,6 +112,170 @@ def unpack_params(params, itemsize, attr_name, buffer, stream=None):
         offset += size
 
 
+class ParamsData(object):
```

Comment:
Could you move all these NCCL-only class and methods to `pure_nccl_communicator.py` ? Then I think you can get rid of re-implementing `_get_nccl_type_id` .

Reply:
Got it. I will move some NCCL related stuff to `pure_nccl_communicator.py`.